cfme/configure/settings.py

```diff
@@ -53,272 +50,339 @@ def __init__(self, description=None, scope=None, days=None, hours=None, timezone
-                         scope=self.scope)
-        fill(self.timeprofile_form, {'description': new_timeprofile.description,
-                              'scope': new_timeprofile.scope},
-             action=self.save_button)
-        end = "saved" if self.appliance.version > '5.7' else "added"
-        flash.assert_success_message('Time Profile "{}" was {}'
-                                     .format(new_timeprofile.description, end))
+                                      scope=self.scope)
+        changed = view.timeprofile_form.fill({
+            'description': self.description + "copy",
+            'scope': self.scope,
+        })
+
+        if changed:
+            view.timeprofile_form.save_button.click()
         return new_timeprofile
 
     def delete(self):
-        navigate_to(self, 'All')
-        row = timeprofile_table.find_row_by_cells({'description': self.description})
-        sel.check(sel.element(".//input[@type='checkbox']", root=row[0]))
-        cfg_btn('Delete selected Time Profiles', invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_success_message(
-            'Time Profile "{}": Delete successful'.format(self.description))
+        view = navigate_to(self, 'Delete')
+        rows = view.tables
+        for row in rows:
+            if row.Description.text == self.description:
+                row[0].check()
+        view.configuration.item_select("Delete selected Time Profiles", handle_alert=True)
 
 
 @navigator.register(Timeprofile, 'All')
 class TimeprofileAll(CFMENavigateStep):
+    VIEW = MySettingsView
     prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
 
     def step(self):
-        tabs.select_tab("Time Profiles")
+        self.view.tabs.time_profile.select()
 
 
 @navigator.register(Timeprofile, 'Add')
 class TimeprofileNew(CFMENavigateStep):
+    VIEW = TimeProfileAddFormView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        cfg_btn('Add a new Time Profile')
+        self.prerequisite_view.configuration.item_select("Add a new Time Profile")
 
 
 @navigator.register(Timeprofile, 'Edit')
 class TimeprofileEdit(CFMENavigateStep):
+    VIEW = TimeProfileAddFormView
+    prerequisite = NavigateToSibling('All')
+
+
+@navigator.register(Timeprofile, 'Copy')
+class TimeprofileCopy(CFMENavigateStep):
+    VIEW = TimeProfileAddFormView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        timeprofile_table.click_cell("description", self.obj.description)
+        rows = self.prerequisite_view.tables
+        rows[0][0].check()
+        self.prerequisite_view.configuration.item_select("Copy selected Time Profile")
 
 
-class Visual(Updateable, Navigatable):
+@navigator.register(Timeprofile, 'Delete')
+class TimeprofileDelete(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+    VIEW = MySettingsView
+    prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
```

Comment:
**REQUIRED**
What's going on here?

Reply:
Will make the changes, sorry for this mistake.